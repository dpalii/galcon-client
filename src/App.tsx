import './App.css';
import { ConnectModal } from './components/ConnectModal/ConnectModal';
import { Game } from './components/Game/Game';
import { useContext, useState } from 'react';
import { SocketContext } from './contexts/SocketContext';
import { GameDetails, IncomingEvents, OutgoingEvents } from './types';

function App() {
  const socket = useContext(SocketContext);
  const [gameId, setGameId] = useState<string | null>(null);
  // const [players, setPlayers] = useState<string[]>([]);
  const handleConnect = (gameId: string) => {
    socket.emit(OutgoingEvents.JOIN_GAME, gameId, (response: GameDetails) => {
      console.log(response.gameId);
      setGameId(response.gameId);
      // setPlayers([response.player1, response.player2]);
    });
  }
  const handleNewGame = () => {
    socket.emit(OutgoingEvents.CREATE_NEW_GAME, (response: string) => {
      console.log(response);
      setGameId(response);
      // setPlayers([]);
    });
  }
  socket.on(IncomingEvents.PLAYER_JOINED, (...args) => {
    console.log(args);
  });

  return (
    <div className="App">
      {
        gameId
          ? <Game gameId={gameId} /> 
          : <ConnectModal onConnect={handleConnect} onCreateNewGame={handleNewGame} />
      }
    </div>
  );
}

export default App;
