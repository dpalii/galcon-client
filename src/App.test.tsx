import { render, unmountComponentAtNode } from 'react-dom';
import React from 'react';
import { act } from 'react-dom/test-utils';
import App from './App';

let container: any = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('<ConnectModal>', () => {
  it('should render component', () => {
    act(() => {
      render(<App />, container);
    });

    expect(container).toMatchSnapshot();
  });
});
