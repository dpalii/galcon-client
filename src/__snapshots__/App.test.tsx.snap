// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectModal> should render component 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="backdrop"
    >
      <div
        class="modal"
      >
        <h2
          class="heading"
        >
          USER CONFIGURATION
        </h2>
        <form>
          <div
            class="input-wrapper"
          >
            <label
              for="name"
            >
              Name
            </label>
            <input
              id="name"
              name="name"
              value=""
            />
          </div>
          <div
            class="input-wrapper"
          >
            <label
              for="color"
            >
              Color
            </label>
            <input
              id="color"
              name="color"
              type="color"
              value="#0000ff"
            />
          </div>
        </form>
        <p
          class="error"
        />
        <button
          class="btn mb w-100"
          id="create-button"
        >
          CREATE GAME
        </button>
        <button
          class="btn mb w-100"
          id="join-button"
        >
          JOIN BY CODE
        </button>
        <button
          class="btn w-100"
          id="open-button"
        >
          OPEN LOBBY LIST
        </button>
      </div>
    </div>
  </div>
</div>
`;
